KN.KloudIdentity.Mapper/MapperCore/IntegrationMethods/AS400Integration.cs

```diff
@@ -0,0 +1,329 @@
+        {
+            { "username", GetValueFromResource(schema, resource, "Username") },
+            { "description", GetValueFromResource(schema, resource, "Description") },
+            { "userClass", userClassValue },
+            { "identifier", GetValueFromResource(schema, resource, "Identifier") }
+        };
+
+        return await Task.FromResult(payload);
+    }
+
+    private string GetValueFromResource(IList<AttributeSchema> schema, Core2EnterpriseUser resource, string destinationField)
+    {
+        var field = schema.FirstOrDefault(x => x.DestinationField.Replace(urnPrefix, string.Empty) == destinationField)?.SourceValue
+                    ?? throw new ArgumentNullException(destinationField, $"{destinationField} not configured in attribute mappings.");
+        var value = JSONParserUtilV2<Core2EnterpriseUser>.ReadProperty(resource, field)?.ToString();
+
+        if (string.IsNullOrEmpty(value))
+        {
+            throw new ArgumentNullException(destinationField, $"{destinationField} is empty.");
+        }
+
+        return value ?? string.Empty;
+    }
+
+    public async Task ProvisionAsync(dynamic payload, AppConfig appConfig, string correlationID, CancellationToken cancellationToken = default)
+    {
+        await ValidatePayloadAsync(payload, correlationID, cancellationToken);
+
+        var basicAuth = await GetAuthenticationAsync(appConfig, SCIMDirections.Outbound, default);
+
+        var apiPath = appConfig.IntegrationDetails!.TrimEnd('/') + "/api/USERS";
+
+        var requestPayload = new
+        {
+            correlationId = correlationID,
+            requestPayload = payload
+        };
+
+        string jsonStringPayload = JsonConvert.SerializeObject(requestPayload);
+
+        var as400RequestMessage = new AS400RequestMessage(
+            apiPath,
+            basicAuth.Username,
+            basicAuth.Password,
+            jsonStringPayload
+        );
+
+        var responseMessage = await SendMessage(correlationID, as400RequestMessage, OperationTypes.Create, cancellationToken);
+
+        if (responseMessage == null || responseMessage?.IsError == true)
+        {
+            throw new ApplicationException($"Error occurred while creating the user: {responseMessage?.ErrorMessage}");
+        }
+    }
+
+    private async Task<StagingQueueResponseMessage> SendMessage(string correlationID, AS400RequestMessage as400RequestMessage, OperationTypes operationType, CancellationToken cancellationToken)
+    {
+        StagingQueueRequestMessage request = new(
+            correlationID,
+            as400RequestMessage,
+            HostTypes.AS400,
+            operationType
+        );
+
+        var encryptedMessage = EncryptWithPrivateKey(request);
+
+        var responseMessage = await _reqStagQueuePublisher.SendAsync(encryptedMessage, correlationID, operationType, cancellationToken);
+        // @ToDo: Decrypt the response message
+        // var result = DecryptWithPublicKey(responseMessage);
+
```

Comment:
Can we remove this commented line?

Reply:
This is a TODO task. Please help us  remember it in the future.

Reply:
Okay noted